modules/ROOT/pages/dw-test-asserts.adoc

```diff
@@ -0,0 +1,828 @@
+----
+
+
+=== beOneOf
+
+==== beOneOf(expected: Array<Any>): Matcher
+
+Validates that the value is contained in the given Array.
+
+===== Parameters
+
+[%header, cols="1,3"]
+|===
+| Name   | Description
+| `expected` | The array of possible elements
+|===
+
+===== Example
+
+Asserts that the value is either `1` or "A Text" or `true`.
+
+====== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+import dw::tests::Asserts
+---
+1 must beOneOf([1, "A Text", true])
+----
+
+
+=== beString
+
+==== beString(): Matcher
+
+Validates that a given value is of type String.
+
+===== Example
+
+This example shows how to validate that a value is of type `String`.
+
+====== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+import dw::tests::Asserts
+---
+"A Text" must beString()
+----
+
+
+=== contain
+
+==== contain(expected: String): Matcher<String>
+
+Validates that the asserted String contains the given String.
+
+===== Parameters
+
+[%header, cols="1,3"]
+|===
+| Name   | Description
+| `expected` | The expected text to be contained
+|===
+
+===== Example
+
+Expects the value to contain the `String` "ex".
+
+====== Source
+[source,DataWeave, linenums]
+----
+%dw 2.0
+import dw::tests::Asserts
+---
+"A Text" must contain("ex")
+----
+
+==== contain(expected: Any): Matcher<Array<Any>>
+
+Validates that the asserted Array contains the given value.
+
+===== Parameters
+
+[%header, cols="1,3"]
+|===
+| Name   | Description
+| `expected` | The expected value to be contained
+|===
+
+===== Example
+
+Expects the `Array` value to contain the `Number` 1.
+
+====== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+import dw::tests::Asserts
+---
+[1, "A Text", true] must contain(1)
+----
+
+
+=== eachItem
+
+==== eachItem(matcher: Matcher<Any>): Matcher<Array<Any>>
+
+Validates that each item of the array satisfies the given matcher.
+
+===== Parameters
+
+[%header, cols="1,3"]
+|===
+| Name   | Description
+| `matcher` | The matcher to apply to all the elements
+|===
+
+===== Example
+
+Expects all the elements in the `Array` to be a `Number`.
+
+====== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+import dw::tests::Asserts
+---
+[1,2,3] must eachItem(beNumber())
+----
+
+
+=== endWith
+
+==== endWith(expected: String): Matcher<String>
+
+Validates that the asserted String ends with the given String.
+
+===== Parameters
+
+[%header, cols="1,3"]
+|===
+| Name   | Description
+| `expected` | Suffix of the 'String'
+|===
+
+===== Example
+
+Expects the `String` to end with "xt".
+
+====== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+import dw::tests::Asserts
+---
+"A Text" must endWith("xt")
+----
+
+
+=== equalTo
+
+==== equalTo(expected: Any, equalToConfig: { unordered?: Boolean } = {}): Matcher<Any>
+
+Validates that a value is equal to another one.
+
+===== Parameters
+
+[%header, cols="1,3"]
+|===
+| Name   | Description
+| `expected` | The expected value
+| `equalToConfig` | Configuration of how to compare them
```

Comment:
```suggestion
| `equalToConfig` | Configuration of how to compare them.
```

Reply:
I won't touch that above to avoid providing a bad description, but the wording is odd.